### PR TITLE
 Add CSP and remove inline styles.

### LINKED ISSUE #36: Content Security Policy
#### As a security-minded developer, I want to obviate all cross site scripting (XSS) attacks.

A strong content security policy can be used to prevent XSS and other
attacks. With many things security related, it is a good idea to disallow
everything and then allow things only after questioning whether or not they
should be allowed (a whitelist policy).

The most restrictive content security policy is:

    Content-Security-Policy: default-src 'none';

However this does not permit any stylesheets, images, nor javascript sources
all of which we want to be able to serve up locally, but not inline. The
following accomplishes that task:

    Content-Security-Policy: default-src 'none'; script-src 'self'; img-src 'self'; style-src 'self';

Acceptance criteria:
- [x] Every request outputs the above content security policy header.
- [x] There are no content security policy violations in chrome's console.
- [x] The user interface remains the same.

Reference material:
- https://www.owasp.org/index.php/Content_Security_Policy_Cheat_Sheet

### PR DESCRIPTION
**Pull request by HaominZhang**
Closes #36

### CODE REVIEW

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -2,4 +2,9 @@ class ApplicationController < ActionController::Base
   # Prevent CSRF attacks by raising an exception.
   # For APIs, you may want to use :null_session instead.
   protect_from_forgery with: :exception
+  before_filter :set_header
+
+  def set_header
+    response.headers['Content-Security-Policy'] = "default-src 'self'; script-src 'self'; img-src *; style-src 'self'"
```
Comment (bboe) [inline app/controllers/application_controller.rb]: We talked about this already but I wanted to put it here for posterity. How can you accomplish this while keeping `"default-src 'none'"`?

File: app/controllers/application_controller.rb [hunk 2]
```diff
@@ -2,4 +2,9 @@ class ApplicationController < ActionController::Base
   # Prevent CSRF attacks by raising an exception.
   # For APIs, you may want to use :null_session instead.
   protect_from_forgery with: :exception
+  before_filter :set_header
```
Comment (bboe) [inline app/controllers/application_controller.rb]: We also talked about this topic. How can you set default response headers without using a before_filter?

File: app/controllers/application_controller.rb (hunk 1 above)
Comment (HaominZhang) [inline app/controllers/application_controller.rb]: Will fix it.

File: app/controllers/application_controller.rb (hunk 2 above)
Comment (HaominZhang) [inline app/controllers/application_controller.rb]: Sure, I will do some research and fix it.

**Comment by HaominZhang** [comment]
@bboe Just updated, please take a look at it.

**Comment by bboe** [comment]
Looks great :+1:

**Comment by HaominZhang** [comment]
Great! I will merge it!

**Pull request merged**